### PR TITLE
DOC Document SiteTree form field scaffolding

### PR DESCRIPTION
**Pull request by GuySartorelli**
Also updates best practices around scaffolding form fields.

## Issue
- https://github.com/silverstripe/silverstripe-cms/issues/2767

### CODE REVIEW

File: en/08_Changelogs/6.0.0.md [hunk 1]
```diff
@@ -70,6 +71,110 @@ All other models used `SearchableDropdownField` for `has_one` relations and `Gri
...
+      - 'FixedHeight'
+      - 'FixedWidth'
+      - 'BottomContent'
+      - 'AlternateContent'
+
+SilverStripe\UserForms\Model\UserDefinedForm:
+  scaffold_cmsfields_settings:
+    restrictRelations:
+      - 'EmailRecipients'
+```
```
Comment (ChloeSartorelli) [inline en/08_Changelogs/6.0.0.md:180]: This yaml snippet is pretty long - if you want I can put it inside a collapsed `<details>` block.

Comment (emteknetnz) [reply en/08_Changelogs/6.0.0.md:180]: Yeah collapsible would be good.

File: en/08_Changelogs/6.0.0.md [hunk 2]
```diff
@@ -70,6 +71,110 @@ All other models used `SearchableDropdownField` for `has_one` relations and `Gri
 |[`Link`](api:SilverStripe\LinkField\Models\Link)|[`LinkField`](api:SilverStripe\LinkField\Form\LinkField)|[`MultiLinkField`](api:SilverStripe\LinkField\Form\MultiLinkField)|No change|
 |[`BlogCategory`](api:SilverStripe\Blog\Model\BlogCategory)|No change|[`TagField`](api:SilverStripe\TagField\TagField)|[`TagField`](api:SilverStripe\TagField\TagField)|
 |[`BlogTag`](api:SilverStripe\Blog\Model\BlogTag)|No change|[`TagField`](api:SilverStripe\TagField\TagField)|[`TagField`](api:SilverStripe\TagField\TagField)|
+|[`Recipient`](api:SilverStripe\UserForms\Model\Recipient)|No change|Changed which `GridfieldComponent` classes are used|Changed which `GridfieldComponent` classes are used|
+
+### `SiteTree` uses form field scaffolding {#sitetree-scaffolding}
+
+[`SiteTree::getCMSFields()`](api:SilverStripe\CMS\Model\SiteTree::getCMSFields()) used to create its form fields from scratch, without calling `parent::getCMSFields()`. This meant that all subclasses of `SiteTree` (i.e. all of your `Page` classes) had to explicitly define all form fields.
+
+`SiteTree::getCMSFields()` now uses same form field scaffolding that all other `DataObject` subclasses use.
```
Comment (emteknetnz) [inline en/08_Changelogs/6.0.0.md]:
```suggestion
`SiteTree::getCMSFields()` now uses the same form field scaffolding that all other `DataObject` subclasses use.
```

**Review by emteknetnz — CHANGES_REQUESTED** [review]

File: en/08_Changelogs/6.0.0.md (hunk 2 above)
Comment (ChloeSartorelli) [reply en/08_Changelogs/6.0.0.md]: Done

File: en/08_Changelogs/6.0.0.md (hunk 1 above)
Comment (ChloeSartorelli) [reply en/08_Changelogs/6.0.0.md:180]: Done

**Review by emteknetnz — APPROVED** [review]

**Pull request merged**